provider/provider.go

```diff
@@ -133,3 +139,62 @@ func GetProviderClient(config lib.Config) (Client, error) {
 		return nil, fmt.Errorf("unknown and invalid provider client")
 	}
 }
+
+// TODO where should this go
+type DependencyCondition struct {
+	Upperbound string
+	Lowerbound string
+	Name       string
+
+	Client Client
+}
+
+func (dc DependencyCondition) Evaluate(log logr.Logger, ctx engine.ConditionContext) (engine.ConditionResponse, error) {
+	resp := engine.ConditionResponse{}
+	deps, err := dc.Client.GetDependencies()
+	if err != nil {
+		return resp, err
+	}
+	var matchedDep *dependency.Dep
+	for _, dep := range deps {
+		if dep.Name == dc.Name {
+			matchedDep = &dep
+			break
+		}
+	}
+	if matchedDep == nil {
+		return resp, nil
+	}
+
+	depVersion, err := version.NewVersion(matchedDep.Version)
+	if err != nil {
+		return resp, err
+	}
+
+	constraintPieces := []string{}
+	if dc.Lowerbound != "" {
+		constraintPieces = append(constraintPieces, "> "+dc.Lowerbound)
+	}
+	if dc.Upperbound != "" {
+		constraintPieces = append(constraintPieces, "< "+dc.Upperbound)
+	}
+	constraints, err := version.NewConstraint(strings.Join(constraintPieces, ", "))
+	if err != nil {
+		return resp, err
+	}
+
+	resp.Matched = constraints.Check(depVersion)
+	resp.Incidents = []engine.IncidentContext{engine.IncidentContext{
+		FileURI: matchedDep.Location,
```

Comment:
I think location means something other than what I thought it did

Reply:
we should probably change that because it's overloaded, and I think tracking where a dependency came from is probably something we'll want to do as well

Reply:
https://github.com/konveyor/analyzer-lsp/issues/73